### PR TITLE
PROD-905 PROD-782 Fix to remove private activity from latest update

### LINKED ISSUE #1615: Latest Activities Widget shows private friend's activity
**Describe the bug**
When a user posted an update when privacy = "only me", that update is shown in the widget "Latest Updates".

**To Reproduce**
1. Login with User1 and create an activity with privacy "Only Me"
2. Disconnect User1
3. Login with User2 and have a look to the widget Latest Updates

=> You will see "User1 posted an update" even if the update is not visible to user 2

In my case, User1 and User2 are connected.

**Expected behavior**
Do not list activities not visible to current logged in user.

**Screenshots**
In the screenshot below, you can see on the Latest updates widget that "Camille posted an update". But when I access the profile of Camille, you can see that there's no activity.
Camille has created an activity with privacy = "only me". That's why Didier can't see that activity.

image

**Support ticket links**
If applicable, add HelpScout link or ticket number where the issue was originally reported.

**Jira issue** : [PROD-905]

[PROD-905]: https://buddyboss.atlassian.net/browse/PROD-905?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### LINKED ISSUE #2644: BB Latest activities widget not shows the private group feed updates even when logged in, whereas the activity page shows
**Describe the bug**
BB Latest activities widget not shows the private group feed updates even when logged in, whereas the activity page shows

**To Reproduce**
1. Add the latest activities to any page and then go to private group and create a feed
2. Check the activity in the widget, will not show
3. Check the activity on the news feed page it will show

**Expected behavior**
For the private type feeds it should display if the logged in user is authorized

**Screenshots**

**Environment**
Client as well as dev site

**Support ticket links**
If applicable, add HelpScout link or ticket number where the issue was originally reported.
https://secure.helpscout.net/conversation/1577275276/154541

**Jira issue** : [PROD-782]

[PROD-782]: https://buddyboss.atlassian.net/browse/PROD-782?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### PR DESCRIPTION
**Pull request by ngolwalkar**
### Changes proposed in this Pull Request:

Fixes #1615
Fixes #2644

### How to test the changes in this Pull Request:

1. Login with User1 and create an activity with privacy "Only Me"
2. Disconnect User1
3. Login with User2 and have a look at the widget Latest Updates

### Proof Screenshots or Video

https://www.loom.com/share/4570bc9b21fb4b6c9c856e18493dff8a

### CODE REVIEW

File: src/bp-core/bp-core-filters.php [hunk 1]
```diff
@@ -144,6 +144,34 @@ function bp_core_exclude_pages( $pages = array() ) {
...
+	
+	if ( ( $key = array_search( 'onlyme', $privacy ) ) !== false ) {
+		unset($privacy[$key]);
+	}
+	
+	return apply_filters( 'bp_core_exclude_onlyme_privacy', $privacy );
+	
+}
+add_filter( 'bb_activity_privacy_values', 'bp_core_exclude_onlyme_privacy' );
+
```
Comment (chetansatasiya) [inline src/bp-core/bp-core-filters.php:174]: @ngolwalkar No need for this filters

File: src/bp-templates/bp-nouveau/includes/activity/widgets.php [hunk 2]
```diff
@@ -120,6 +120,7 @@ public function widget( $args, $instance ) {
 			'action'       => join( ',', $type ),
 			'primary_id'   => 0,
 			'secondary_id' => 0,
+			'privacy' 	   => apply_filters( 'bb_activity_privacy_values', $privacy ),
```
Comment (chetansatasiya) [inline src/bp-templates/bp-nouveau/includes/activity/widgets.php:123]: @ngolwalkar No need for this as well just pass the `'scope'        => 'all',` to `'scope'        => bp_activity_default_scope( 'all' ),`

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
Fix the changes requested

**Review by chetansatasiya — APPROVED** [review]
LGTM

**Comment by KartikSuthar** [comment]
Closing the PR as proper fix over here: https://github.com/buddyboss/buddyboss-platform/pull/2913

**Pull request closed without merging**